src/components/Search/index.tsx

```diff
@@ -84,31 +102,74 @@ const handleSearchResult = ({
   }
 }
 
-const handleInputFocus = (searchBarEditable: boolean, inputElement: any, dispatch: AppDispatch) => {
-  if (searchBarEditable) {
-    const input: HTMLInputElement = inputElement.current!
-    input.focus()
-    input.addEventListener('focusout', () => {
-      dispatch({
-        type: ComponentActions.UpdateHeaderSearchEditable,
-        payload: {
-          searchBarEditable: false,
-        },
-      })
-    })
-  }
-}
-
 const Search = ({ dispatch, hasBorder, content }: { dispatch: AppDispatch; hasBorder?: boolean; content?: string }) => {
+  const defaultSearchSuggestions: SearchSuggestion[] = []
   const [searchValue, setSearchValue] = useState(content || '')
+  const [searchSuggestions, setSearchSuggestions] = useState(defaultSearchSuggestions)
+  const [expandSuggestions, setExpandSuggestions] = useState(false)
   const [placeholder, setPlaceholder] = useState(SearchPlaceholder)
-  const inputElement = useRef(null)
+  const [maximumBlockHeight, setMaximumBlockHeight] = useState(0)
+  const inputElement = useRef<HTMLInputElement>(null)
   const { components } = useContext(AppContext)
   const { searchBarEditable } = components
 
   useEffect(() => {
-    handleInputFocus(searchBarEditable, inputElement, dispatch)
-  }, [searchBarEditable, dispatch])
+    const tipBlockNumberInterval = setInterval(async () => {
+      const { attributes } = await fetchTipBlockNumber()
+      setMaximumBlockHeight(attributes.tip_block_number)
+    }, BLOCK_POLLING_TIME)
+
+    return () => clearInterval(tipBlockNumberInterval)
+  }, [])
+
+  useEffect(() => {
+    if (inputElement.current) {
+      inputElement.current.focus()
+    }
+  }, [searchBarEditable])
```

Comment:
`searchBarEditable ` does not exist in useEffect, why set `searchBarEditable ` to dependence array.